File: src_a_(C++)/Test5-11.c
//Проект Странник-Модула Для Windows 32, тестовая программа
//Группа тестов 5:ПРОЦЕДУРЫ
//Тест номер    11:ВЫЗОВ ФУНКЦИИ ВНУТРИ ФУНКЦИИ

include Win32

char s[15];

int i;

void pr1(int p1,int p2) {
  p1=p2;
}

int pr2(int p1,int p2) {
  p1=p2;
  return(0);
}

int pr3() {
  MessageBox(0,"Ok","Ok",0);
  return(0);
}

void main() {
  pr1(8,pr2(2,pr3()));
  pr1(8,pr2(2,MessageBox(0,"Ok","Ok",0)));
}

File: src_a_(Pascal)/Test5-11.pas
//Проект Странник-Модула Для Windows 32, тестовая программа
//Группа тестов 5:ПРОЦЕДУРЫ
//Тест номер    11:ВЫЗОВ ФУНКЦИИ ВНУТРИ ФУНКЦИИ
program Test5_11;
uses Win32;

var s:string[15];

var i:integer;

procedure pr1(p1,p2:integer);
begin
  p1:=p2;
end;

function pr2(p1,p2:integer):integer;
begin
  p1:=p2;
  p2:=0;
end;

function pr3():integer;
begin
  MessageBox(0,"Ok","Ok",0);
  pr3:=0;
end;

begin
  pr1(8,pr2(2,pr3()));
  pr1(8,pr2(2,MessageBox(0,"Ok","Ok",0)));
end.

